### PR TITLE
test that both processes are executed

### PR DESCRIPTION
**Pull request by crq**

### CODE REVIEW

File: test/run_test.js [hunk 1]
```diff
@@ -1,22 +1,24 @@
...
+import { expect } from 'chai'
+import { spy } from 'sinon'
 
+describe('lib/run', () => {
   describe('runTasks', () => {
-    const tasks = [
-      'this:thing',
-      'that:thing'
-    ]
+    const tasks = ['dummy:1', 'dummy:2']
```
Comment (ianmcnally) [inline test/run_test.js:8]: not very descriptive tasks names

:grimacing:

File: test/run_test.js [hunk 2]
```diff
@@ -1,22 +1,24 @@
...
       return runTasks(tasks)
     })
 
-    it('executes an `npm run` for each task', () => {
+    after(() => {
+      childProcess.exec.restore()
     })
 
+    it('executes an `npm run` for each task', () => {
+      expect(childProcess.exec).to.have.been.calledWith(`npm run dummy:1`)
```
Comment (ianmcnally) [inline test/run_test.js:20]: could iterate over tasks and assert, like `tasks.forEach(task => expect()...with(`npm run${task}`)`

File: package.json [hunk 3]
```diff
@@ -12,10 +12,13 @@
     "prepublish": "npm run build",
     "start": "node $npm_package_main",
     "test": "mocha --compilers js:babel-register $npm_package_config_tests --require $npm_package_config_test_helpers",
+    "test:debug": "npm test -- --debug",
```
Comment (ianmcnally) [inline package.json:15]: what does the debug flag give you?

File: test/run_test.js [hunk 4]
```diff
@@ -1,22 +1,24 @@
...
+describe('lib/run', () => {
   describe('runTasks', () => {
-    const tasks = [
-      'this:thing',
-      'that:thing'
-    ]
+    const tasks = ['dummy:1', 'dummy:2']
 
     before(() => {
+      spy(childProcess, 'exec')
```
Comment (ianmcnally) [inline test/run_test.js:11]: How do you feel about spying and actually opening the processes? I'm kind of torn. I like not having to overly mock it, but I don't like that the unit test depends on tasks in package.json and start a process.

File: test/run_test.js (hunk 2 above)
Comment (crq) [inline test/run_test.js:20]: Sometimes I like being explicit in tests :smile: but yea we could do that.

File: package.json (hunk 3 above)
Comment (crq) [inline package.json:15]: Opps... meant to take this out - was trying something out.

**Comment by ianmcnally** [comment]
solid work here @crq. i went ahead with the suggestions i made in 575ef6d, which were simple tweaks on what you did. thanks!

**Comment by ianmcnally** [comment]
Also decided to make it out:  438c96c

If you put a wrong task in, the child process would kill the test watch. The child process would also output onto the test screen. Not committed to that way of mocking the streams, but it's a decent first pass.

**Pull request closed without merging**